webextension/background/analytics.js

```diff
@@ -1,26 +1,33 @@
-/* globals main, auth */
+/* globals main, auth, catcher */
 
 window.analytics = (function () {
   let exports = {};
 
   exports.sendEvent = function (action, label) {
-    let eventCategory = "addon";
-    let url = main.getBackend() + "/event";
-    let req = new XMLHttpRequest();
-    req.open("POST", url);
-    req.setRequestHeader("content-type", "application/json");
-    req.onload = () => {
-      if (req.status >= 300) {
-        console.warn("Event gave non-2xx response:", req.status);
-      }
-    };
-    // FIXME: add cdX and other details from req.js
-    req.send(JSON.stringify({
-      deviceId: auth.getDeviceId(),
-      event: eventCategory,
-      action,
-      label
-    }));
+    return new Promise((resolve, reject) => {
+      let eventCategory = "addon";
+      let url = main.getBackend() + "/event";
+      let req = new XMLHttpRequest();
+      req.open("POST", url);
+      req.setRequestHeader("content-type", "application/json");
+      req.onload = () => {
+        if (req.status >= 300) {
+          let exc = new Error("Bad response from POST /event");
+          exc.status = req.status;
+          exc.statusText = req.statusText;
+          reject(exc);
+        } else {
+          resolve();
+        }
+      };
+      // FIXME: add cdX and other details from req.js
```

Comment:
Could you just add the other details as part of this PR? Or if not, file a bug and include the bug number in the FIXME comment